Doc/library/importlib.rst

```diff
@@ -465,6 +466,72 @@ ABC hierarchy::
            The import machinery now takes care of this automatically.
 
 
+.. class:: ResourceReader
+
+    An :term:`abstract base class` for :term:`loaders <loader>`
+    representing a :term:`package` to provide the ability to read
+    *resources*.
+
+    From the perspective of this class, a *resource* is a binary
+    artifact that is shipped within the package that this loader
+    represents. Typically this is something like a data file that
+    lives next to the ``__init__.py`` file of the package. The
+    purpose of this class is to help abstract out the accessing of
+    such data files so that it does not matter if the package and
+    its data file(s) are stored in a e.g. zip file versus on the
+    file system.
+
+    For any of methods of this class, a *resource* argument is
+    expected to be a :term:`file-like object` which represents
+    conceptually just a file name. This means that no subdirectory
+    paths should be included in the *resource* argument. This is
+    because the location of the package that the loader is for acts
+    as the "directory". Hence the metaphor for directories and file
+    names is packages and resources, respectively. This is also why
+    instances of this class are expected to directly correlate to
+    a specific package (instead of potentially representing multiple
+    packages or a module).
+
+    .. versionadded:: 3.7
+
+    .. abstractmethod:: open_resource(resource)
+
+        Returns an opened, :term:`file-like object` for binary reading
+        of the *resource*.
+
+        If the resource cannot be found, :exc:`FileNotFoundError` is
+        raised.
+
+    .. abstractmethod:: resource_path(resource)
+
+        Returns the file system path to the *resource*.
+
+        If the resource does not concretely exist on the file system,
+        raise :exc:`FileNotFoundError`.
+
+    .. abstractmethod:: is_resource(path)
```

Comment:
In the [standalone package](http://importlib-resources.readthedocs.io/en/latest/abc.html) I called this *name* so that it doesn't necessarily imply a path (which could include slashes).